partialWaveFit/pwaLikelihood.cc

```diff
@@ -1089,8 +1098,23 @@ pwaLikelihood<complexT>::addAmplitude(const amplitudeMetadata& meta)
 		         << "Aborting..." << endl;
 		return false;
 	}
-
-	const string waveName = meta.objectBaseName();
+	if (metas.size() == 0) {
+		printErr << "no amplitudeMetadata pointer given" << endl;
+		return false;
+	}
+	for (size_t nMeta = 0; nMeta < metas.size(); ++nMeta) {
+		if (not metas[nMeta]) {
+			printErr << "amplitudeMetadata metas[" << nMeta << "] not valid. Aborting..." << endl;
+			return false;
+		}
+	}
+	const string waveName = metas[0]->objectBaseName();
+	for (size_t nMeta = 0;nMeta < metas.size(); ++nMeta) {
```

Comment:
nMeta could start at 1, also, there is a space missing after the first semicolon.